### PR TITLE
8302870: More information needed from failures in vmTestbase ThreadUtils.waitThreadState

### PR DESCRIPTION
**Pull request by kevinjwalls**
Test update to make failures such as JDK-8076494 more informative.

Waiting for a thread to change state: give more time (to distinguish a real deadlock from some other delay), and log the thread stackframes when the expected state change is not observed.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8302870](https://bugs.openjdk.org/browse/JDK-8302870): More information needed from failures in vmTestbase ThreadUtils.waitThreadState

### Reviewers
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12661/head:pull/12661` \
`$ git checkout pull/12661`

Update a local copy of the PR: \
`$ git checkout pull/12661` \
`$ git pull https://git.openjdk.org/jdk pull/12661/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12661`

View PR using the GUI difftool: \
`$ git pr show -t 12661`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12661.diff">https://git.openjdk.org/jdk/pull/12661.diff</a>

</details>

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/nsk/monitoring/share/thread/ThreadUtils.java [hunk 1]
```diff
@@ -218,20 +228,33 @@ public static void threadDump(Log log, ThreadInfo[] tinfos) {
                 }
         }
 
+        // Most uses of waitThreadState usually succeed without retries.
+        // Sleep time increased to help avoid spurious failures.
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/vmTestbase/nsk/monitoring/share/thread/ThreadUtils.java:232]: "increased" doesn't mean anything outside this PR as you can't tell the value has changed.

Comment (dholmes-ora) [reply test/hotspot/jtreg/vmTestbase/nsk/monitoring/share/thread/ThreadUtils.java:232]: Also any comment on sleep time belongs after line 233.

File: test/hotspot/jtreg/vmTestbase/nsk/monitoring/share/thread/ThreadUtils.java [hunk 2]
```diff
@@ -218,20 +228,33 @@ public static void threadDump(Log log, ThreadInfo[] tinfos) {
...
         public final static int waitThreadStateRetries = 10;
-        public final static long waitThreadStateSleepTime = 100;
+        public final static long waitThreadStateSleepTime = 1000;
 
         public static void waitThreadState(Thread thread, Thread.State state) {
                 int retries = 0;
                 long ctime = System.currentTimeMillis();
+                long ctime2 = 0;
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/vmTestbase/nsk/monitoring/share/thread/ThreadUtils.java:239]: So ctime -> startTime; ctime2 -> elapsedTime ?

File: test/hotspot/jtreg/vmTestbase/nsk/monitoring/share/thread/ThreadUtils.java [hunk 3]
```diff
@@ -218,20 +228,33 @@ public static void threadDump(Log log, ThreadInfo[] tinfos) {
...
+                            // Show the thread, and fail with Exception showing retries and time taken.
+                            ctime2 = System.currentTimeMillis() - ctime;
+                            System.err.println("waitThreadState: problem thread: " + thread);
+                            printStackTrace(System.err, thread);
+                            throw new TestFailure("Thread " + thread + " with current state " + thread.getState() + " did not reach state "
+                                                  + state + " with number of retries: " + retries + ", time: " + ctime2);
+                        }
                         try {
-                                Thread.sleep(waitThreadStateSleepTime);
+                            Thread.sleep(waitThreadStateSleepTime);
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/vmTestbase/nsk/monitoring/share/thread/ThreadUtils.java:251]: This file is using an indent of 8

File: test/hotspot/jtreg/vmTestbase/nsk/monitoring/share/thread/ThreadUtils.java [hunk 4]
```diff
@@ -218,20 +228,33 @@ public static void threadDump(Log log, ThreadInfo[] tinfos) {
...
 
         public static void waitThreadState(Thread thread, Thread.State state) {
                 int retries = 0;
                 long ctime = System.currentTimeMillis();
+                long ctime2 = 0;
                 while (thread.getState() != state) {
-                        if (retries++ > waitThreadStateRetries)
-                                throw new TestFailure("Thread " + thread + " with current state " + thread.getState() + " did not reach state " + state + " with number of  retries: " + retries + ", time: " + (System.currentTimeMillis() - ctime));
+                        if (retries++ > waitThreadStateRetries) {
+                            // Failed to see desired state, give info to help diagnose the isuse.
```
Comment (dholmes-ora) [inline test/hotspot/jtreg/vmTestbase/nsk/monitoring/share/thread/ThreadUtils.java:242]: This file is using an indent of 8. Looks odd to use 4 for the modified code only. Maybe you could fix all the indents before integration?

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
Approach is fine in principle but a few minor nits.

Thanks.

**Comment by kevinjwalls** [comment]
OK thanks David, nits taken care of I think.  (I'm not falling for the trap of changing all the indenting so nobody can see from the diff what I'm changing here, so I suppressed my instincts and used 8 spaces.)

**Review by dholmes-ora — APPROVED** [review]
Update looks good - thanks.

I have no issue if you change the file indents after the PR is approved and then integrate - as it is only whitespace change. :)

**Review by sspitsyn — APPROVED** [review]
Looks good.
Thanks,
Serguei

**Comment by kevinjwalls** [comment]
Thanks David and Serguei!
(leaving whitespace unification alone for now... 8-) )

**Pull request closed without merging**